backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -206,7 +207,8 @@ public String getNeededData(EvaluationContext evalContext) {
                 && entries.elementAt(0) instanceof SyncEntry
                 && ((SyncEntry)entries.elementAt(0)).getSyncPost().isRelevant(evalContext)) {
             return SessionFrame.STATE_SYNC_REQUEST;
-        } else if (entries.size() > 1 || !entries.elementAt(0).getCommandId().equals(currentCmd)) {
+        } else if (entries.size() > 1 ||
+                (!entries.isEmpty() && !entries.elementAt(0).getCommandId().equals(currentCmd))) {
```

Comment:
Adding `entries.isEmpty()` prevents a index out of range crash, but it might take us into undefined territory in terms of session nav...